apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/ErrorReporting/ErrorReportingTest.cs

```diff
@@ -0,0 +1,271 @@
+                response = await requestTask2;
+                Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+                response = await requestTask3;
+                Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+                response = await requestTask4;
+                Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+
+                var errorEvents = _polling.GetEvents(startTime, testId, 4);
+                Assert.Equal(4, errorEvents.Count());
+
+                var exceptionEvents = errorEvents.Where(e => e.Message.Contains("ThrowsException"));
+                Assert.Equal(3, exceptionEvents.Count());
+                foreach (var errorEvent in exceptionEvents)
+                {
+                    VerifyErrorEvent(errorEvent, testId, "ThrowsException");
+                }
+
+                var argumentExceptionEvents = errorEvents.Where(e => e.Message.Contains("ThrowsArgumentException"));
+                Assert.Single(argumentExceptionEvents);
+                VerifyErrorEvent(argumentExceptionEvents.First(), testId, "ThrowsArgumentException");
+            }
+        }
+
+        [Fact]
+        public async Task LogsThrownInHttpMessageHandler()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            using (TestServer server = TestServer.Create<ErrorReportingTestApplication>())
+            using (var client = server.HttpClient)
+            {
+                var response = await server.HttpClient.GetAsync($"handler/{testId}");
+                Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+
+                var errorEvent = _polling.GetEvents(startTime, testId, 1).Single();
+                VerifyErrorEvent(errorEvent, testId, "SendAsync");
+            }
+        }
+
+        /// <summary>
+        /// Checks that an <see cref="ErrorEvent"/> contains valid data.
+        /// </summary>
+        /// <param name="errorEvent">The event to check.</param>
+        /// <param name="testId">The id of the test.</param>
+        /// <param name="functionName">The name of the function the error occurred in.</param>
+        private void VerifyErrorEvent(ErrorEvent errorEvent, string testId, string functionName)
+        {
+            Assert.Equal(ErrorReportingController.Service, errorEvent.ServiceContext.Service);
+            Assert.Equal(ErrorReportingController.Version, errorEvent.ServiceContext.Version);
+
+            Assert.Contains(functionName, errorEvent.Message);
+            Assert.Contains(testId, errorEvent.Message);
+
+            Assert.Equal(HttpMethod.Get.Method, errorEvent.Context.HttpRequest.Method);
+            Assert.False(string.IsNullOrWhiteSpace(errorEvent.Context.HttpRequest.Url));
+            //Assert.True(errorEvent.Context.HttpRequest.ResponseStatusCode >= 200);
```

Comment:
Should this be uncommented?

Reply:
I actually need to remove that. The System.Web.Http.ExceptionHandling.ExceptionLoggerContext that we are receiving on Google.Cloud.Diagnostics.AspNet.ErrorReportingExceptionLogger.Log has the Response set to null on ExceptionContext and that's where we get the ResponseStatusCode from. This again might be because of the self hosted pipeline, I want to try and see if I can add integration tests for an IIS hosted application.